ivy/functional/ivy/experimental/layers.py

```diff
@@ -1235,23 +1235,12 @@ def embedding(
     ivy.utils.assertions.check_equal(
```

Comment:
Given that this assertion is defined in the `ivy` implementation, this will only be used when no backend is set, we'd need to use an assertion like this in the backend implementations as well if we'd like to check for the same whena. backend has been set.